lib/dynflow/exporters/abstract.rb

```diff
@@ -2,103 +2,41 @@ module Dynflow
-      def initialize(world = nil, options = {})
-        @world   = world
+      def initialize(options = {})
         @options = options
-        @index   = {}
-      end
-
-      # Add the execution_plan to the index, thus queueing it for exporting
-      def add(execution_plan)
-        @index[execution_plan.id] = {
-          :plan   => execution_plan,
-          :result => nil
-        }
-        self
-      end
-
-      # Add provided id to the index
-      # The id will be resolved to the execution plan
-      # in the #resolve_ids method along with all the other
-      # index entries whose :plan is nil
-      def add_id(execution_plan_id)
-        @index[execution_plan_id] = {
-          :plan   => nil,
-          :result => nil
-        }
-        self
-      end
-
-      # The same as #add, but takes Array[ExecutionPlan]
-      def add_many(execution_plans)
-        execution_plans.each { |plan| add plan }
-        self
-      end
-
-      # The same as #add_id, but takes Array[String]
-      def add_many_ids(execution_plan_ids)
-        execution_plan_ids.each { |plan_id| add_id plan_id }
-        self
       end
 
-      # Processes the entries in index and freezes the index and all entries
-      def finalize
-        return self if @index.frozen?
-        resolve_ids
-        @index.each do |key, value|
-          @index[key].update(:result => export(value[:plan]))
-          @index[key].freeze
-        end
-        @index.freeze
-        self
-      end
-
-      # Generally put all the entries' results into an array
-      def result
-        return @result if @result
-        finalize # In case someone forgot to finalize
-        @result = @index.map { |_key, value| value[:result] }
-      end
-
-      # Export index of all entries
-      def export_index
-        @index.keys
-      end
-
-      private
-
       # Implement this method in sub-classes to provide the real exporting functionality
       # Transforms an execution plan to its exported representation
       def export(plan)
         raise NotImplementedError
       end
 
-      # Selects all the entries from the index whose :plan is nil
-      # Loads execution plans for those ids from the database
-      def resolve_ids
-        ids = @index.select { |_key, value| value[:plan].nil? }.keys
-        return if ids.empty?
-        @world.persistence.find_execution_plans(:filters => { :uuid => ids }).each do |plan|
-          @index[plan.id].update(:plan => plan)
-        end
+      # Used as a filename suffix when streaming to a file
+      def filetype
+        raise NotImplementedError
+      end
+
+      # Used when streaming collection, should be a 3-tuple containing
+      #   what is printed before items
+      #   what is printed between the items
+      #   what is printed after the items
+      def brackets
+        []
       end
 
+      # def self.export_execution_plan_id(world, execution_plan_id, options = {})
```

Comment:
Removed, exporting one single task is much simpler now so it doesn't need a helper method